### PR TITLE
Stochastic Gradient Descent.

### PR DESCRIPTION
**Pull request by kellertuer**
This PR provides a few basic stochastic gradient descent methods, remaining steps are

* [x] classic stochastic gradient descent
* [x] with momentum (slightly adapted to tangent spaces)
* [x] with averaging (slightly adapted to tangent spaces)
* [x] Documentation
* [x] Testing
* [x] a small tutorial

### CODE REVIEW

**Comment by kellertuer** [comment]
While AdaGrad seems more challenging that  though (for an half-a-day-project), I found a good way to include also Nesterov-type acceleration to gradient descent.

**Comment by kellertuer** [comment]
AdaGrad seems until now only be available in coordinates. That i s possible in this framework for sure, but I'll leave that for a later PR.

File: docs/src/solvers/stochastic_gradient_descent.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# [Gradient Descent](@id StochasticGradientDescentSolver)
+
+```@meta
+CurrentModule = Manopt
+```
+
+```@docs
+  stochastic_gradient_descent
```
Comment (matbesancon) [inline docs/src/solvers/stochastic_gradient_descent.md:8]: The space shouldn't be needed before the function name?
```suggestion
stochastic_gradient_descent
```

Comment (kellertuer) [reply docs/src/solvers/stochastic_gradient_descent.md]: Thanks for noticing.

Comment (matbesancon) [reply docs/src/solvers/stochastic_gradient_descent.md]: no problem, I was browsing the PR for my own instruction :)

**Pull request merged**